wrappers/android/tools/camera/src/main/java/com/intel/realsense/camera/SettingsActivity.java

```diff
@@ -415,6 +428,43 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
         String filePath = data.getStringExtra(getString(R.string.intent_extra_file_path));
         switch (requestCode){
             case OPEN_FILE_REQUEST_CODE:{
+
+                // firmware file source location
+                Uri uri = data.getData();
+                Log.d(TAG, "uri: " + uri.getPath());
+
+                try {
+                    ContentResolver contentResolver = getContentResolver();
+
+                    // firmware file in app local storage, for example,
+                    // /storage/emulated/0/Android/data/com.intel.realsense.camera/files/Download
+                    File downloadDir = getExternalFilesDir(Environment.DIRECTORY_DOWNLOADS);
```

Comment:
I remember that the folder was /storage/emulated/0/Android/data/com.intel.realsense.camera/files/realsense/firmware.
Why do you change it?

Reply:
this is temporary working directory for the unsigned fw, use the standard download folder, transparent to the user. avoid mixing with the custom folder that contains the signed firmware.

Reply:
So, in order to test it, the unsigned firmware should be located in /storage/emulated/0/Android/data/com.intel.realsense.camera/files/Download - correct?

Reply:
@remibettan no, the unsigned firmware file can be in any accessible folder mounted on the system, then user pick the file through the file browser. The file will be copied to this app local folder and flash to the device. The process is transparent, user do not need to know the location of this folder.

Reply:
@gwen2018 sounds great.